Conflict with glibc 2.35+ macro.

Build error due to conflict with `glibc 2.35` macros

```
filament/libs/image/src/ImageSampler.cpp:41:17: error: expected unqualified-id
constexpr float M_PIf = float(filament::math::F_PI);
                ^
/usr/include/math.h:1168:17: note: expanded from macro 'M_PIf'
# define M_PIf          3.14159265358979323846f /* pi */
                        ^
1 error generated.
```

conflict is caused by recently added macros in glibc 2.35, is there a way around this issue ? since downgrading glibc doesn't seems to be a good idea.

**To Reproduce**
run filament `build.sh` script, on `clang-13` and `glibc 2.35+`


**Expected behavior**
filament release build for android and desktop

**Logs**
```
[23/1345] Building CXX object libs/image/CMakeFiles/image.dir/src/ImageSampler.cpp.o
FAILED: libs/image/CMakeFiles/image.dir/src/ImageSampler.cpp.o 
/usr/bin/clang++ -DFILAMENT_DRIVER_SUPPORTS_VULKAN -DFILAMENT_SUPPORTS_OPENGL -DFILAMENT_SUPPORTS_X11 -DFILAMENT_SUPPORTS_XCB -DFILAMENT_SUPPORTS_XLIB -I/home/machine/workspace/ar/filament/libs/image/include -I/home/machine/workspace/ar/filament/libs/math/include -I/home/machine/workspace/ar/filament/libs/utils/include -I/home/machine/workspace/ar/filament/third_party/robin-map/tnt/.. -stdlib=libc++ -std=c++17 -fstrict-aliasing -Wno-unknown-pragmas -Wno-unused-function -Wno-deprecated-declarations -stdlib=libc++ -fPIC -fcolor-diagnostics -fvisibility=hidden -O3 -DNDEBUG -fomit-frame-pointer -ffunction-sections -fdata-sections -ffast-math -Wno-deprecated-register -MD -MT libs/image/CMakeFiles/image.dir/src/ImageSampler.cpp.o -MF libs/image/CMakeFiles/image.dir/src/ImageSampler.cpp.o.d -o libs/image/CMakeFiles/image.dir/src/ImageSampler.cpp.o -c /home/machine/workspace/ar/filament/libs/image/src/ImageSampler.cpp
/home/machine/workspace/ar/filament/libs/image/src/ImageSampler.cpp:41:17: error: expected unqualified-id
constexpr float M_PIf = float(filament::math::F_PI);
                ^
/usr/include/math.h:1168:17: note: expanded from macro 'M_PIf'
# define M_PIf          3.14159265358979323846f /* pi */
                        ^
1 error generated.
[32/1345] Building CXX object third_party/libgtest/tnt/CMakeFiles/gtest.dir/__/googletest/src/gtest-all.cc.o
ninja: build stopped: subcommand failed.

```

**Desktop**
 - OS: Manjaro Linux
 - GPU: NVIDIA GTX 1650

--- a/libs/image/src/ImageSampler.cpp
+++ b/libs/image/src/ImageSampler.cpp
@@ -32,13 +32,13 @@ using namespace image;
 
 namespace {
 
+using namespace filament::math;
+
 struct FilterFunction {
     float (*fn)(float) = nullptr;
     float boundingRadius = 1;
     bool rejectExternalSamples = true;
 };
-
-constexpr float M_PIf = float(filament::math::F_PI);
 
 const FilterFunction Box {
     .fn = [](float t) { return t <= 0.5f ? 1.0f : 0.0f; },
@@ -50,7 +50,7 @@ const FilterFunction Nearest { Box.fn, 0.0f };
 const FilterFunction Gaussian {
     .fn = [](float t) {
         if (t >= 2.0) return 0.0f;
-        const float scale = 1.0f / std::sqrt(0.5f * M_PIf);
+        const float scale = 1.0f / std::sqrt(0.5f * f::PI);
         return std::exp(-2.0f * t * t) * scale;
     },
     .boundingRadius = 2
@@ -86,7 +86,7 @@ const FilterFunction Mitchell {
 // Not bothering with a fast approximation since we cache results for each row.
 float sinc(float t) {
     if (t <= 0.00001f) return 1.0f;
-    return std::sin(M_PIf * t) / (M_PIf * t);
+    return std::sin(f::PI * t) / (f::PI * t);
 }
 
 const FilterFunction Lanczos {

